### PR TITLE
conmon: Add container id to syslog

### PR DESCRIPTION
**Pull request by mrunalp**
Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by runcom** [comment]
LGTM
/test all

**Comment by wking** [comment]
Since the macros do not check for a useful `opt_cid`, you may want to move [this check][1] up before the [`opt_restore_path && opt_exec` check][2] possibly use a different error-logging mechanism for the `opt_cid` check.

In general, I think logging to syslog for option-parsing errors is a bit odd (before we `fork` into a daemon, the caller can check our exit code for these errors), so an alternative would be to set a `enable_syslog` (or some such) flag which was initially `0` and gets set to `1` post-`fork`.

[1]: https://github.com/kubernetes-incubator/cri-o/blob/e2bb6aaa7931607820b85b8020f47d19efb4e0a3/conmon/conmon.c#L1241-L1242
[2]: https://github.com/kubernetes-incubator/cri-o/blob/e2bb6aaa7931607820b85b8020f47d19efb4e0a3/conmon/conmon.c#L1238-L1239

File: conmon/conmon.c [hunk 1]
```diff
@@ -1235,11 +1235,12 @@ int main(int argc, char *argv[])
 		exit(0);
 	}
 
+	if (opt_cid == NULL)
+		fprintf(stderr, "Container ID not provided. Use --cid");
```
Comment (wking) [inline conmon/conmon.c]: You need to `exit(EXIT_FAILURE);` in this block too.

Comment (mrunalp) [reply conmon/conmon.c]: Fixed.

File: conmon/conmon.c [hunk 2]
```diff
@@ -1235,11 +1235,14 @@ int main(int argc, char *argv[])
 		exit(0);
 	}
 
+	if (opt_cid == NULL) {
+		fprintf(stderr, "Container ID not provided. Use --cid");
```
Comment (wking) [inline conmon/conmon.c]: With the direct `fprintf`, you need to explicitly add a trailing `\n`.

Comment (rhatdan) [reply conmon/conmon.c]: Does this mean every caller to conmon now must exepcy the --cid option?

Comment (wking) [reply conmon/conmon.c]: > Does this mean every caller to conmon now must exepcy the --cid option?

Yup, but that's [not new][1].

[1]: https://github.com/kubernetes-incubator/cri-o/pull/1475/files#diff-caf7c5c19d8bb76309f55ccce30a55afL1241

Comment (mrunalp) [reply conmon/conmon.c]: Yeah, we always need a container id.

**Comment by mrunalp** [comment]
@runcom @rhatdan Tests are green here.

File: conmon/conmon.c (hunk 2 above)
Comment (rhatdan) [reply conmon/conmon.c]: Then why not make it an argument, rather then an option. Options should be Optional.  :^(

**Pull request merged**